sdk/dotnet/Pulumi/Core/InputList.cs

```diff
@@ -57,11 +57,21 @@ private InputList(Output<ImmutableArray<T>> values)
 
         public void Add(params Input<T>[] inputs)
         {
-            // Make an Output from the values passed in, mix in with our own Output, and combine
-            // both to produce the final array that we will now point at.
-            _outputValue = Output.Concat(_outputValue, Output.All(inputs));
+            _outputValue = Concat(inputs);
         }
 
+        public void Add(InputList<T> inputs)
```

Comment:
I would expect this to be called AddRange or Concat but I'm guessing you've named it Add explicitly so you can use it in initializer statements. As I said above I think that's a little unusual compared to the normal dotnet collections.

Reply:
Yes, this is more about making the collection initializers more user friendly than about following other dotnet conventions. These collections are used extensively by resources so it helps a lot with inline resource instantiation and therefore simplifies dotnet code using pulumi libraries.

Reply:
Changed to delegate to `AddRange` so intention is clearer.